### PR TITLE
Keepout areas for cris_new_area_fixed

### PR DESCRIPTION
**Pull request by morpheus1820**

### CODE REVIEW

File: app/maps/cris_new_area_fixed_keepout_mask.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+image: cris_new_area_fixed_keepout_mask.pgm
+resolution: 0.050000
+origin: [-100.000000, -100.000000, 0.000000]
```
Comment (ste93) [inline app/maps/cris_new_area_fixed_keepout_mask.yaml]:
```suggestion
origin: [-13.1, -7.62, 0]
```

File: app/navigation2/conf/cris_new_area_fixed_keepout_params.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+costmap_filter_info_server:
+  ros__parameters:
+    use_sim_time: true
```
Comment (ste93) [inline app/navigation2/conf/cris_new_area_fixed_keepout_params.yaml]:
```suggestion
```

File: app/navigation2/conf/cris_new_area_fixed_keepout_params.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    type: 0
+    filter_info_topic: "/costmap_filter_info"
+    mask_topic: "/keepout_filter_mask"
+    base: 0.0
+    multiplier: 1.0
+filter_mask_server:
+  ros__parameters:
+    use_sim_time: true
```
Comment (ste93) [inline app/navigation2/conf/cris_new_area_fixed_keepout_params.yaml]:
```suggestion
```

File: app/navigation2/conf/nav2_params.yaml [hunk 4]
```diff
@@ -173,7 +173,11 @@ controller_server:
       RotateToGoal.slowing_factor: 5.0
       RotateToGoal.lookahead_time: -1.0
 
-controller_server_rclcpp_node:
+      filters: ["keepout_filter"]
+      keepout_filter:
+        plugin: "nav2_costmap_2d::KeepoutFilter"
+        enabled: True
+        filter_info_topic: "/costmap_filter_info"controller_server_rclcpp_node:
```
Comment (ste93) [inline app/navigation2/conf/nav2_params.yaml]:
```suggestion
controller_server_rclcpp_node:
```

File: app/navigation2/conf/nav2_params.yaml [hunk 5]
```diff
@@ -316,6 +325,10 @@ global_costmap:
         cost_scaling_factor: 2.5
         inflate_unknown: false
         inflate_around_unknown: false
+      keepout_filter:
```
Comment (ste93) [inline app/navigation2/conf/nav2_params.yaml:325]:
```suggestion
      filters: ["keepout_filter"]
      keepout_filter:
```

File: app/navigation2/launch/costmap_filter_info.launch.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+    # Declare the launch arguments
+    declare_namespace_cmd = DeclareLaunchArgument(
+        'namespace',
+        default_value='',
+        description='Top-level namespace')
+
+    declare_use_sim_time_cmd = DeclareLaunchArgument(
+        'use_sim_time',
+        default_value='true',
```
Comment (ste93) [inline app/navigation2/launch/costmap_filter_info.launch.py]:
```suggestion
        default_value='false',
```

**Review by ste93 — CHANGES_REQUESTED** [review]

**Comment by ste93** [comment]
also add in `app/navigation2/scripts/Navigation_ROS2_R1SN003.xml` the following code at line 92
```
   <module>
      <name>ros2</name>
      <parameters>launch launch/costmap_filter_info.launch.py params_file:=conf/cris_new_area_fixed_keepout_params.yaml mask:=/home/user1/tour-guide-robot/app/maps/cris_new_area_fixed_keepout_mask.yaml</parameters>
      <workdir>/home/user1/tour-guide-robot/app/navigation2/</workdir>
      <node>console</node>
   </module>
```

**Comment by elandini84** [comment]
Hi @morpheus1820, can you apply the changes suggested by @ste93 so we can proceed with rebasing and merging the PR?

**Comment by morpheus1820** [comment]
> Hi @morpheus1820, can you apply the changes suggested by @ste93 so we can proceed with rebasing and merging the PR?

done

**Comment by elandini84** [comment]
> Hi @morpheus1820, can you apply the changes suggested by @ste93 so we can proceed with rebasing and merging the PR?

Ok, can you rebase it on ros2 branch? The branch is currently 15 commits behind [origin/ros2](https://github.com/hsp-iit/tour-guide-robot)

**Pull request closed without merging**